velox/expression/tests/ExprTest.cpp

```diff
@@ -1751,6 +1752,84 @@ TEST_F(ExprTest, ifWithConstant) {
   EXPECT_EQ(true, result->as<ConstantVector<bool>>()->valueAt(0));
 }
 
+// Make sure that switch do set nulls for rows that are not evaluated by the
+// switch due to a throw when the output type is not fixed primitive.
+TEST_F(ExprTest, switchSetNullsForThrowIndices) {
+  // Build an input with c0 column having nulls at odd row index.
+  registerFunction<TestingAlwaysThrowsFunction, bool, int64_t>(
+      {"always_throws"});
+  registerFunction<TestingThrowsAtOddFunction, bool, int64_t>({"throw_at_odd"});
+
+  auto eval = [&](const std::string& input, auto inputRow) {
```

Comment:
can you add a comment why we arnt wrapping it in try instead, so that we can verify that nulls are added otherwise try would add them for us